cram_pr2/cram_pr2_environment_manipulation/src/trajectories.lisp

```diff
@@ -126,43 +183,114 @@
+
+    (alexandria:switch
+        (object-type :test (lambda (?type ?super-type)
+                             (prolog:prolog
+                              `(man-int:object-type-subtype
+                                ,?super-type ,?type))))
+      (:container-prismatic
+       (make-prismatic-trajectory object-transform arm action-type grasp-pose opening-distance))
+      (:container-revolute
+       (make-revolute-trajectory object-transform arm action-type grasp-pose opening-distance))
+      (T (error "Unsupported container-type: ~a." object-type)))))
+
+(defun make-prismatic-trajectory (object-transform arm action-type
+                                  grasp-pose opening-distance)
+  "Return a list of `man-int::traj-segment' representing a trajectory to open a
+container with prismatic joints."
+  (mapcar (lambda (label transforms)
                 (man-int:make-traj-segment
                  :label label
-                 :poses (list
-                         (funcall object-to-standard-gripper->base-to-particular-gripper
-                                  transform))))
-                (list
-                 :reaching
-                 :grasping
-                 action-type
-                 :retracting)
-                (list
-                 (cram-tf:translate-transform-stamped
-                  grasp-pose :x-offset *drawer-handle-pregrasp-x-offset*)
-                 grasp-pose
-                 (cram-tf:translate-transform-stamped
-                  grasp-pose :x-offset opening-distance)
-                 (cram-tf:translate-transform-stamped
-                  grasp-pose :x-offset (+ opening-distance *drawer-handle-retract-offset*)))))))
+                 :poses (mapcar (man-int:make-object-to-standard-gripper->base-to-particular-gripper-transformer
+                                 object-transform arm)
+                                transforms)))
+              (list
+               :reaching
+               :grasping
+               action-type
+               :retracting)
+              (list
+               (list (cram-tf:translate-transform-stamped
+                      grasp-pose :x-offset *drawer-handle-pregrasp-x-offset*))
+               (list grasp-pose)
+               (list (cram-tf:translate-transform-stamped
+                      grasp-pose :x-offset opening-distance))
+               (list (cram-tf:translate-transform-stamped
+                      grasp-pose :x-offset (+ opening-distance *drawer-handle-retract-offset*))))))
 
-(defmethod man-int:get-action-trajectory ((action-type (eql :opening))
-                                           arm
-                                           grasp
-                                           objects-acted-on
-                                           &key
-                                             opening-distance)
-  (make-trajectory action-type arm objects-acted-on opening-distance))
+(defun make-revolute-trajectory (object-transform arm action-type
```

Comment:
This returns the same trajectory for the opening and closing action, whereas they are actually supposed to be inverted. I do not see anywhere in the code the inversion. Have you tested your feature with a closing action?

Reply:
Ah, you invert the opening-distance, i see. Please ignore the above comment.